Octokit.Tests.Integration/Clients/RepositoryContentsClientTests.cs

```diff
@@ -0,0 +1,133 @@
+            }
+
+            [IntegrationTest]
+            public async Task GetsDirectoryContent()
+            {
+                var github = new GitHubClient(new ProductHeaderValue("OctokitTests"))
+                {
+                    Credentials = Helper.Credentials
+                };
+
+                var contents = await github
+                    .Repository
+                    .Content
+                    .GetContents("octokit", "octokit.net", "Octokit");
+
+                Assert.True(contents.Count > 2);
+                Assert.Equal(ContentType.Dir, contents.First().Type);
+            }
+        }
+
+        [IntegrationTest]
+        public async Task CrudTest()
+        {
+            var client = new GitHubClient(new ProductHeaderValue("OctokitTests"))
+            {
+                Credentials = Helper.Credentials
+            };
+            Repository repository = null;
+            try
+            {
+                var fixture = client.Repository.Content;
+                var repoName = Helper.MakeNameWithTimestamp("source-repo");
+                repository = await client.Repository.Create(new NewRepository { Name = repoName, AutoInit = true });
+
+                var file = await fixture.CreateFile(
+                    repository.Owner.Login,
+                    repository.Name,
+                    "somefile.txt",
+                    new CreateFileRequest("Test commit", "Some Content"));
+                Assert.Equal("somefile.txt", file.Content.Name);
+
+                var contents = await fixture.GetContents(repository.Owner.Login, repository.Name, "somefile.txt");
+                string fileSha = contents.First().Sha;
+                Assert.Equal("Some Content", contents.First().Content);
+
+                var update = await fixture.UpdateFile(
+                    repository.Owner.Login,
+                    repository.Name,
+                    "somefile.txt",
+                    new UpdateFileRequest("Updating file", "New Content", fileSha));
+                Assert.Equal("somefile.txt", update.Content.Name);
+
+                contents = await fixture.GetContents(repository.Owner.Login, repository.Name, "somefile.txt");
+                Assert.Equal("New Content", contents.First().Content);
+                fileSha = contents.First().Sha;
+
+                await fixture.DeleteFile(
+                    repository.Owner.Login,
+                    repository.Name,
+                    "somefile.txt",
+                    new DeleteFileRequest("Deleted file", fileSha));
+
+                await Assert.ThrowsAsync<NotFoundException>(
+                    async () => await fixture.GetContents(repository.Owner.Login, repository.Name, "somefile.txt"));
+            }
+            finally
+            {
+                if (repository != null)
+                    client.Repository.Delete(repository.Owner.Login, repository.Name);
```

Comment:
We've got a couple of different ways to dispose of repositories when we create them in the scope of an integration test (mostly I believe it's around using `IDisposable` in tests to use xUnit hooks.

Not a blocker on shipping this feature, but I'd like to unify these: https://github.com/octokit/octokit.net/issues/655